diff --git a/dist/properties/index.d.ts b/dist/properties/index.d.ts
index 7c26ef4f04373c846ba4e0d2d76480be14e13aa3..f73875219bcb05c3ea48f36de4f20f2e18188531 100644
--- a/dist/properties/index.d.ts
+++ b/dist/properties/index.d.ts
@@ -16,7 +16,11 @@ import { CaretProperties } from '../caret.js';
 import { alignmentXMap, alignmentYMap, ColorRepresentation, VisibilityProperties } from '../utils.js';
 import { SelectionProperties } from '../selection.js';
 import { LayerInSectionIdentifier } from './layer.js';
-export type BaseOutProperties = YogaProperties & PanelProperties & ZIndexProperties & TransformProperties & ScrollbarProperties & PanelGroupProperties & VisibilityProperties & PointerEventsProperties & ListenersProperties & EventHandlersProperties & TextAlignProperties & AppearanceProperties & FontFamilyProperties & GlyphProperties & CaretProperties & SelectionProperties & SizeProperties & AnchorProperties & CursorProperties & IdProperties & ComponentDefaultsProperties;
+export type NameUserDataProperties = {
+    name?: string;
+    userData?: Record<string, any>;
+};
+export type BaseOutProperties = YogaProperties & PanelProperties & ZIndexProperties & TransformProperties & ScrollbarProperties & PanelGroupProperties & VisibilityProperties & PointerEventsProperties & ListenersProperties & EventHandlersProperties & TextAlignProperties & AppearanceProperties & FontFamilyProperties & GlyphProperties & CaretProperties & SelectionProperties & SizeProperties & AnchorProperties & CursorProperties & IdProperties & ComponentDefaultsProperties & NameUserDataProperties;
 export type CursorProperties = {
     cursor?: string;
 };
