diff --git a/dist/build.js b/dist/build.js
index 07c945a190851415c7090ce65b5705499c7c4fde..105024486fd5b6a094ab24d7636efe4d95209fc2 100644
--- a/dist/build.js
+++ b/dist/build.js
@@ -12,7 +12,7 @@ export function build(Component, name = Component.name) {
         const renderContext = useRenderContext();
         const args = useMemo(() => [undefined, undefined, { renderContext }], [renderContext]);
         const outProps = useSetup(ref, props, args);
-        return jsx(`vanilla${name}`, { ref, children, ...outProps });
+        return jsx(`vanilla${name}`, { ref, children, ...outProps, name: props.name, userData: props.userData });
     });
 }
 export function useRenderContext() {
